dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -301,28 +301,30 @@ public void Dispose()
     private static Dictionary<string, ISKFunction> ImportSkill(object skillInstance, string skillName, ILogger logger)
     {
         MethodInfo[] methods = skillInstance.GetType().GetMethods(BindingFlags.Static | BindingFlags.Instance | BindingFlags.Public);
-        logger.LogTrace("Importing skill name: {0}. Potential methods found: {1}", skillName, methods.Length);
-
+        logger.LogTrace($"Importing skill name: {skillName}. Potential methods found: {methods.Length}");
+    
         // Filter out non-SKFunctions and fail if two functions have the same name
         Dictionary<string, ISKFunction> result = new(StringComparer.OrdinalIgnoreCase);
         foreach (MethodInfo method in methods)
         {
             if (method.GetCustomAttribute<SKFunctionAttribute>() is not null)
             {
                 ISKFunction function = SKFunction.FromNativeMethod(method, skillInstance, skillName, logger);
-                if (result.ContainsKey(function.Name))
+                try
+                {
+                    result.Add(function.Name, function);
+                }
+                catch (ArgumentException)
                 {
                     throw new KernelException(
                         KernelException.ErrorCodes.FunctionOverloadNotSupported,
                         "Function overloads are not supported, please differentiate function names");
                 }
-
-                result.Add(function.Name, function);
             }
         }
-
-        logger.LogTrace("Methods imported {0}", result.Count);
-
+    
+        logger.LogTrace($"Methods imported {result.Count}");
```

Comment:
Revert this too.  Same reason as above. The old code enabled structured logging.